Rocket.Chat/Controllers/Chat/ChatDataController.swift

```diff
@@ -193,11 +202,11 @@ final class ChatDataController {
         return (indexPaths, removedIndexPaths)
     }
 
-    func update(_ message: Message) -> Int {
+    func update(_ message: Message, asyncUpdate: (() -> Void)?) -> Int {
```

Comment:
Actually, this `asyncUpdate` is being called on the main thread, right? Should we change to something like `completion`?

Reply:
:thinking:
```swift
        DispatchQueue.global(qos: .background).async {
            let finalText = NSMutableAttributedString(attributedString: text.transformMarkdown())
            finalText.trimCharacters(in: .whitespaces)
            finalText.highlightMentions(mentions, username: username)
            finalText.highlightChannels(channels)
            self.cache.setObject(finalText, forKey: key)
            asyncUpdate?()
        }
```